### PR TITLE
Bug 1916021: Fix infinite loop when a CSV replacement chain contains a cycle.

### PR DESCRIPTION
**Pull request by benluddy**

### CODE REVIEW

File: pkg/controller/operators/olm/operator.go [hunk 1]
```diff
@@ -1923,6 +1926,9 @@ func (a *Operator) getReplacementChain(in *v1alpha1.ClusterServiceVersion, csvsI
 		csvsInChain[current] = struct{}{}
 	}
 	for prev != nil && *prev != "" {
+		if _, ok := csvsInChain[*next]; ok {
```
Comment (hasbro17) [inline pkg/controller/operators/olm/operator.go:1929]: Should this be checking the previous csv in the chain?
```suggestion
		if _, ok := csvsInChain[*prev]; ok {
```

Comment (benluddy) [reply pkg/controller/operators/olm/operator.go]: Yes, good catch. Added a test case to cover this mistake and fixed it. Thanks!

**Comment by benluddy** [comment]
> @benluddy: The following test **failed**, say `/retest` to rerun all failed tests:
> Test name 	Commit 	Details 	Rerun command
> ci/prow/e2e-aws-console-olm 	[8e568ba](https://github.com/operator-framework/operator-lifecycle-manager/commit/8e568baa757942bea4062f591a5b67b03eb863e8) 	[link](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/operator-framework_operator-lifecycle-manager/1966/pull-ci-operator-framework-operator-lifecycle-manager-master-e2e-aws-console-olm/1351919797931085824) 	`/test e2e-aws-console-olm`
> 
> [Full PR test history](https://prow.ci.openshift.org/pr-history?org=operator-framework&repo=operator-lifecycle-manager&pr=1966). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:benluddy).

It looks like https://github.com/openshift/console/pull/7874 is in progress to resolve this test failure.

**Pull request merged**